src/components/InputDataDialog.js

```diff
@@ -1,54 +1,50 @@
 import styled from 'styled-components';
 
 export default function InputDataDialog({updateInput}) {
-    return (
-        <FormInputData onSubmit={handleSubmit}>
-            <label htmlFor="inputfield">Enter your energy consumption - watt/h:</label>
-            <input
-                type="text" min="0" maxLength={20} pattern="([0-9]+)"
-                id="inputfield" name="inputfield" required />
-            <button>Submit</button>
-        </FormInputData>
-
-    );
-
-    function handleSubmit(inputEvent) {
-        inputEvent.preventDefault();
-        const inputData = inputEvent.target;
-        const inputDataValue = inputData.elements.inputfield.value.trim();
-
-        updateInput(inputDataValue);
-
-        inputData.reset();
-    }
+  return (
+    <FormInputData onSubmit={handleSubmit}>
+      <label htmlFor="inputfield">Enter your energy consumption - watt/h:</label>
+      <input type="text" min="0" pattern="[0-9]{1,10}" id="inputfield" name="inputfield" required />
+      <button>Submit</button>
+    </FormInputData>
+  );
+
+  function handleSubmit(inputEvent) {
+    inputEvent.preventDefault();
+    const inputData = inputEvent.target;
+    const inputDataValue = inputData.elements.inputfield.value.trim();
+
+    updateInput(inputDataValue);
```

Comment:
I would recommend to use more specific names. _What_ are you updating exactly? A better name would probably be something like: `updateEnergyConsumption`.